Source/Core/DolphinQt/Settings/BroadbandAdapterSettingsDialog.cpp

```diff
@@ -0,0 +1,106 @@
+    : QDialog(parent)
+{
+  this->m_bba_type = bba_type;
+  InitControls();
+  setLayout(main_layout);
+}
+
+void BroadbandAdapterSettingsDialog::InitControls()
+{
+  switch (m_bba_type)
+  {
+  case Type::ETHERNET:
+    // i18n: MAC stands for Media Access Control. A MAC address uniquely identifies a network
+    // interface (physical) like a serial number. "MAC" should be kept in translations.
+    bba_address_label = new QLabel(tr("Enter new Broadband Adapter MAC address:"));
+    bba_address_placeholder = QString::fromStdString("aa:bb:cc:dd:ee:ff");
+    bba_current_address = QString::fromStdString(SConfig::GetInstance().m_bba_mac);
+    bba_description = new QLabel(tr("For setup instructions, <a "
+                                    "href=\"https://wiki.dolphin-emu.org/"
+                                    "index.php?title=Broadband_Adapter\">refer to this page</a>."));
+
+    // i18n: MAC stands for Media Access Control. A MAC address uniquely identifies a network
+    // interface (physical) like a serial number. "MAC" should be kept in translations.
+    bba_window_title = tr("Broadband Adapter MAC Address");
+    break;
+
+  case Type::XLINK_KAI:
+    bba_address_label = new QLabel(tr("Enter IP address of device running the XLink Kai Client:"));
+    bba_address_placeholder = QString::fromStdString("127.0.0.1");
+    bba_current_address = QString::fromStdString(SConfig::GetInstance().m_bba_xlink_ip);
+    bba_description =
+        new QLabel(tr("For setup instructions, <a "
+                      "href=\"https://www.teamxlink.co.uk/wiki/Dolphin\">refer to this page</a>."));
+    bba_window_title = tr("XLink Kai BBA Destination Address");
+    break;
+  }
+
+  setWindowTitle(bba_window_title);
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  m_bba_address_input = new QLineEdit(bba_current_address);
+  m_bba_address_input->setPlaceholderText(bba_address_placeholder);
+
+  auto bba_buttonbox = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel);
+  connect(bba_buttonbox, &QDialogButtonBox::accepted, this,
+          &BroadbandAdapterSettingsDialog::SaveAddress);
+  connect(bba_buttonbox, &QDialogButtonBox::rejected, this,
+          &BroadbandAdapterSettingsDialog::reject);
+
+  bba_description->setTextFormat(Qt::RichText);
+  bba_description->setWordWrap(true);
+  bba_description->setTextInteractionFlags(Qt::TextBrowserInteraction);
+  bba_description->setOpenExternalLinks(true);
+
+  main_layout = new QVBoxLayout();
+  main_layout->addWidget(bba_address_label);
+  main_layout->addWidget(m_bba_address_input);
+  main_layout->addWidget(bba_description);
+  main_layout->addWidget(bba_buttonbox);
+}
+
+void BroadbandAdapterSettingsDialog::SaveAddress()
+{
+  const std::string bba_new_address(StripSpaces(m_bba_address_input->text().toStdString()));
+
+  switch (m_bba_type)
+  {
+  case Type::ETHERNET:
+    if (!std::regex_match(bba_new_address, std::regex("([0-9A-Fa-f]{2}:){5}([0-9A-Fa-f]{2})")))
```

Comment:
Just checking, should this have start/end anchors?
```suggestion
    if (!std::regex_match(bba_new_address, std::regex("^([0-9A-Fa-f]{2}:){5}([0-9A-Fa-f]{2})$")))
```

Reply:
I haven't tried. The current regex (without the anchors) is working as intended, though...

Reply:
std::regex_match only considers matches on the whole string, so the anchors are implied.